### ISSUE TITLE
Photon+Carousel: Add responsive imaging support

### ISSUE THREAD

**Issue by kraftbj** [Automattic/jetpack#3791]
The Carousel loads the full-size image for the image in view, the previous, and the next image in the carousel. This can be really excessive depending on the size of the original images.

We should improve this, likely using srcset values, so the browser can choose something sensible relative to the viewport for images that are large.

cc: @josephscott 
related: #2918

**Comment by josephscott** [comment]
You can see how this plays out from this WebPageTest waterfall - http://www.webpagetest.org/result/160511_JN_2234e36c63cf726274111317c1553819/1/details/ - run from an iPhone 5C ( so it has a retina screen ).

At the bottom are requests for the original versions of three images: the one being viewed, the previous one, and the next one ( in the carousel ).  That is how a mobile page view ends up coming in at 15MB :-(

**Comment by josephscott** [comment]
I've been testing this change:

```php
--- jetpack-carousel.js-orig    2016-05-13 21:52:40.420664000 +0000
+++ jetpack-carousel.js 2016-05-14 03:43:19.268664000 +0000
@@ -949,6 +949,8 @@

            // Give devices with a higher devicePixelRatio higher-res images (Retina display = 2, Android phones = 1.5, etc)
            if ( 'undefined' !== typeof window.devicePixelRatio && window.devicePixelRatio > 1 ) {
+               args.orig_max_width = args.max_width;
+               args.orig_max_height = args.max_height;
                args.max_width  = args.max_width * window.devicePixelRatio;
                args.max_height = args.max_height * window.devicePixelRatio;
            }
@@ -965,9 +967,13 @@
                // args.orig_file doesn't point to a Photon url, so in this case we use args.large_file
                // to return the photon url of the original image.
                var largeFileIndex = args.large_file.lastIndexOf( '?' );
-               var origPhotonUrl = args.large_file;
                if ( -1 !== largeFileIndex ) {
                    origPhotonUrl = args.large_file.substring( 0, largeFileIndex );
+                   // If we have a really large image load a smaller version
+                   // that is closer to the viewable size
+                   if ( args.orig_width > args.max_width || args.orig_height > args.max_height ) {
+                       origPhotonUrl += '?fit=' + args.orig_max_width + '%2C' + args.orig_max_height;
+                   }
                }
                return origPhotonUrl;
            }
```

For large images that we've been testing with ( like 5500 x 4500 ) this loads more reasonable sized versions, even on retina and mobile.

**Comment by jeherve** [comment]
Fixed in #3801
